File: module-contracts/src/main/java/com/beat/contracts/storage/FileStoragePort.java
package com.beat.contracts.storage;

import java.util.List;

public interface FileStoragePort {

	PerformancePresignedUrls issueAllPresignedUrlsForPerformanceMaker(
		String posterImage,
		List<String> castImages,
		List<String> staffImages,
		List<String> performanceImages
	);

	CarouselPresignedUrls issueAllPresignedUrlsForCarousel(List<String> carouselImages);

	BannerPresignedUrl issuePresignedUrlForBanner(String bannerImage);
}

File: module-contracts/src/main/kotlin/com/beat/contracts/storage/FileStoragePort.kt
package com.beat.contracts.storage


interface FileStoragePort {

    fun issueAllPresignedUrlsForCarousel(carouselImages: List<String>): CarouselPresignedUrls

    fun findImageObjectMetadata(imageKey: String): ImageObjectMetadata?

    fun issuePresignedUrlForBanner(bannerImage: String): BannerPresignedUrl
}
